### PR TITLE
Configure assumption exception

### LINKED ISSUE #2267: Allow configuring the order of assumption exception classes
#### Problem

Our project has dependencies on both JUnit5 and TestNG. Most test cases are written by JUnit5.
`Assumptions.assumeThat()` is also used with JUnit5. TestNG is only used for running Reactive Streams TCK.
https://github.com/reactive-streams/reactive-streams-jvm/blob/899f17ffc2927bf89fad73dfde182f5018359bdf/tck/build.gradle#L19
When I tried to upgrade AssertJ from 3.17.1, our JUnit test suits failed with:
```
FlagsTest > epollAvailableOnLinux() FAILED
    org.testng.SkipException: assumption was not met due to: 
    Expecting:
      ...
```
The error seems to be caused by the change of the order of assumption exceptions in https://github.com/assertj/assertj-core/commit/cf65f6a069814abbf478b9fda47db0978e6dfa75.

#### Suggestion

It would be nice to give an option that changes the order of assumption exceptions using a property file or a JVM system property.
```
-Dorg.assertj.assumeption.exceptions=org.opentest4j.TestAbortedException,org.testng.SkipException,org.junit.AssumptionViolatedException
```

### PR DESCRIPTION
**Pull request by joel-costigliola**
#### Check List:
* Fixes #2267
* Unit tests : YES
* Javadoc with a code example (on API only) : YES

### CODE REVIEW

**Comment by joel-costigliola** [comment]
@scordio do you mind having a look at this one? (if you are not too busy)

**Comment by scordio** [comment]
Sure, I'll have a look during the weekend.

File: src/test/java/org/assertj/core/api/EntryPointAssumptions_setPreferredAssumptionException_Test.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import org.assertj.core.configuration.PreferredAssumptionException;
+import org.junit.AssumptionViolatedException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opentest4j.TestAbortedException;
+
+@DisplayName("EntryPoint assumptions setPreferredAssumptionException method")
+class EntryPointAssumptions_setPreferredAssumptionException_Test {
```
[the remark this answers is not in the document]
Comment (joel-costigliola) [reply src/test/java/org/assertj/core/api/EntryPointAssumptions_setPreferredAssumptionException_Test.java:39]: agreed

File: src/main/java/org/assertj/core/configuration/PreferredAssumptionException.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  /**
+   * AssertJ will try to build the exception to throw in this order:
+   * <ol>
+   * <li>{@code org.testng.SkipException} for TestNG (if available in the classpath)</li>
+   * <li>{@code org.junit.AssumptionViolatedException} for JUnit4 (if available in the classpath)</li>
+   * <li>{@code org.opentest4j.TestAbortedException} for JUnit5</li>
+   * </ol> 
+   * If none are available, AssertJ throw an {@link IllegalStateException}.
+   */
+  AUTO_DETECT("try in order org.testng.SkipException, org.junit.AssumptionViolatedException and org.opentest4j.TestAbortedException");
```
[the remark this answers is not in the document]
Comment (joel-costigliola) [reply src/main/java/org/assertj/core/configuration/PreferredAssumptionException.java]: I did it and the result is nicer, I'll you check the updated code.

[the remark this answers is not in the document]
Comment (scordio) [reply src/main/java/org/assertj/core/configuration/PreferredAssumptionException.java]: I'd go one step further, see #2278

**Review by scordio — APPROVED** [review]
LGTM

**Comment by stbischof** [comment]
This closed not merged? @joel-costigliola

**Comment by scordio** [comment]
@stbischof merged in 9690ffc9345c8f22a1dd2d77dddc0bf635005510.

**Pull request closed without merging**